param/parameterized.py

```diff
@@ -874,18 +876,48 @@ def _changed(cls, change):
         return not Comparator.is_equal(change.old, change.new)
 
 
-    @classmethod
-    def _call_watcher(cls, watcher, change):
+
+    def _call_watcher(self_, watcher, change):
         """
         Invoke the given the watcher appropriately given a Change object.
         """
-        if watcher.onlychanged and (not cls._changed(change)):
+        if watcher.onlychanged and (not self_._changed(change)):
             return
-        if watcher.mode == 'args':
+
+        if self_.cls.param._BATCH_WATCH:
+            self_._changes.append(change)
+            self_._watchers.append(watcher)
```

Comment:
Maybe:

```python
if watcher not in self_._watchers:
    self_.watchers.append(watcher)
```

Then you don't have to compute an ordered set later.

Reply:
Good idea...assuming I only ever need a set (which I currently think is true)

Reply:
Addressed in 7c4b4ce.